test/integration/Main.hs

```diff
@@ -290,6 +290,14 @@ testScenarioSolution _ci _em =
               when (shouldCheckBadErrors == CheckForBadErrors) $ noBadErrors g
               verify g
 
+  tutorialHasLog :: GameState -> Assertion
+  tutorialHasLog gs =
+    let baseDevs = gs ^?! baseRobot . equippedDevices
+     in assertBool "Base should have a logger installed!" (not . null $ lookupByName "logger" baseDevs)
+
+  testTutorialSolution t f = testSolution' t f CheckForBadErrors tutorialHasLog
```

Comment:
Sure, but given the order of parameters, you would not gain anything right now.

If we add more conditions, then it might be worth refactoring. :slightly_smiling_face: